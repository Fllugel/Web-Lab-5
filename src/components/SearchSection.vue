<template>
    <div class="ui inverted vertical masthead center aligned segment">
      <div class="ui text container">
        <canvas class="background"></canvas>
        <h1 class="ui inverted header">Find your course</h1>
        <div class="ui fluid big action input"> 
          <input ref="searchInput" type="text" v-model="searchQuery" @keyup.enter="searchCourses" placeholder="Search...">
          <div class="ui green button" @click="searchCourses"><i class="search icon"></i></div>
        </div>
      </div>            
    </div>
  </template>
  
  <script>
  import Particles from 'particlesjs';
  
  export default {
    data() {
      return {
        searchQuery: ''
      };
    },
    mounted() {
      // Initialize particles
      //this.initParticles();
    },
    methods: {
      searchCourses() {
        // Redirect to courses page with search query
        if (this.searchQuery.trim() !== '') {
          this.$router.push({ path: '/courses', query: { search: this.searchQuery } });
        } else {
          // If search query is empty, redirect to courses page without query
          this.$router.push('/courses');
        }
      },
      initParticles() {
        const background = document.querySelector(".background");
        const rect = background.getBoundingClientRect(); // Get dimensions of the background
  
        const minDistance = Math.min(rect.width, rect.height) * 0.8; // Adjust the multiplier as needed for desired spacing
  
        Particles.init({
          selector: ".background",
          color: "#99cd32",
          connectParticles: true,
          sizeVariations: 10,
          maxParticles: 100,
          minDistance: minDistance, // Set the minimum distance based on background size
        });
      }
    }
  }
  </script>
  
  <style scoped>
  /* Search section styles */
  </style>
  