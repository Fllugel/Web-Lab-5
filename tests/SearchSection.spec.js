import { shallowMount } from '@vue/test-utils';
import SearchSection from '@/components/SearchSection.vue';

describe('SearchSection.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SearchSection);
  });

  it('renders the component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('has a search input', () => {
    const input = wrapper.find('input[type="text"]');
    expect(input.exists()).toBe(true);
  });

  it('has a search button', () => {
    const button = wrapper.find('div.ui.green.button');
    expect(button.exists()).toBe(true);
  });

  it('has a canvas element', () => {
    const canvas = wrapper.find('canvas.background');
    expect(canvas.exists()).toBe(true);
  });

  it('renders the header', () => {
    const header = wrapper.find('h1.ui.inverted.header');
    expect(header.text()).toBe('Find your course');
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
