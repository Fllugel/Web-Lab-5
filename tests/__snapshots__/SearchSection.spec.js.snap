// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSection.vue matches the snapshot 1`] = `
"<div class="ui inverted vertical masthead center aligned segment">
  <div class="ui text container"><canvas class="background"></canvas>
    <h1 class="ui inverted header">Find your course</h1>
    <div class="ui fluid big action input"><input type="text" placeholder="Search...">
      <div class="ui green button"><i class="search icon"></i></div>
    </div>
  </div>
</div>"
`;
